livekit/src/room/mod.rs

```diff
@@ -262,9 +262,8 @@ struct RoomInfo {
 
 pub(crate) struct RoomSession {
     rtc_engine: Arc<RtcEngine>,
-    sid_tx: AsyncMutex<Option<oneshot::Sender<()>>>,
-    sid_rx: AsyncMutex<oneshot::Receiver<()>>,
-    sid: RwLock<Option<RoomSid>>,
+    sid_promise: Promise<RoomSid>,
+    sid: AsyncMutex<Option<RoomSid>>,
```

Comment:
Can we avoid having the sid field? Ideally it is already contained inside the promise, so we keep things even simpler.

Reply:
probably not, for two reasons:
- the Promise doesn't keep the value stored anywhere, it just returns from recv when ready (though this could be changed to allow for one-send-multiple-recv)
- even if we do that, though, `Promise<AsyncSid>` is too opaque for stuff like `impl Debug` or `impl From` that need to be synchronous. i could add a `try_result()` method to `Promise<T>` but that would be... hacky. might still be better though.

Reply:
new commit addresses this, lmk if that's better